test/integration/fake_upstream.cc

```diff
@@ -73,22 +73,25 @@ void FakeStream::decodeMetadata(Http::MetadataMapPtr&& metadata_map_ptr) {
 }
 
 void FakeStream::encode100ContinueHeaders(const Http::HeaderMapImpl& headers) {
-  std::shared_ptr<Http::HeaderMapImpl> headers_copy(
+  std::unique_ptr<Http::HeaderMapImpl> headers_copy(
       new Http::HeaderMapImpl(static_cast<const Http::HeaderMap&>(headers)));
-  parent_.connection().dispatcher().post(
-      [this, headers_copy]() -> void { encoder_.encode100ContinueHeaders(*headers_copy); });
+  parent_.connection().dispatcher().post([this, headers = headers_copy.release()]() -> void {
+    encoder_.encode100ContinueHeaders(*headers);
```

Comment:
you have to delete headers here otherwise it will be flagged as a leak.

Reply:
Yeah, already fixed that locally, I'm just waiting for tests to finish to avoid spamming CI.

Reply:
Fixed.